test/sharness/t0021-config.sh

```diff
@@ -126,6 +126,13 @@ test_config_cmd() {
     test_cmp replconfig.json newconfig.json
   '
 
+  # Dynamic keys with dot in their names
+  test_config_cmd_set "--json" "Gateway.PublicGateways[\"some.example.com\"].UseSubdomains" "true"
+  test_expect_success "'ipfs config' output for dynamic keys looks good" '
+    ipfs config Gateway.PublicGateways["some.example.com"].UseSubdomains > dynkey_out &&
+    grep true dynkey_out
+  '
+
```

Comment:
It'd be nice if we could validate that the result is valid JSON, this only proves that our code can read what it writes. Perhaps something like piping `ipfs config show` into `jq` to see that the Gateway JSON is what we expect it to be.

Reply:
Good idea, added in 732adebaf